app/uk/gov/hmrc/gform/formtemplate/Rewriter.scala

```diff
@@ -533,18 +534,71 @@ trait Rewriter {
             )
         }
 
+      def updateTaskSection(section: Section, taskCaption: Option[SmartString]): Section =
+        section match {
+          case s: Section.NonRepeatingPage =>
+            val newCaption: Option[SmartString] = s.page.caption match {
+              case Some(value) => Some(value)
+              case None        => taskCaption
+            }
+            s.copy(
+              page = s.page.copy(
+                includeIf = replaceIncludeIf(s.page.includeIf),
+                fields = replaceFields(s.page.fields),
+                redirects = replaceRedirects(s.page.redirects),
+                confirmation = replaceConfirmation(s.page.confirmation),
+                caption = newCaption
+              )
+            )
+          case s: Section.RepeatingPage =>
+            val newCaption: Option[SmartString] = s.page.caption match {
+              case Some(value) => Some(value)
+              case None        => taskCaption
+            }
+            s.copy(
+              page = s.page.copy(
+                includeIf = replaceIncludeIf(s.page.includeIf),
+                fields = replaceFields(s.page.fields),
+                redirects = replaceRedirects(s.page.redirects),
+                confirmation = replaceConfirmation(s.page.confirmation),
+                caption = newCaption
+              )
+            )
+          case s: Section.AddToList =>
+            val newCaption: Option[SmartString] = s.caption match {
+              case Some(value) => Some(value)
+              case None        => taskCaption
+            }
+            s.copy(
+              includeIf = replaceIncludeIf(s.includeIf),
+              repeatsUntil = replaceIncludeIf(s.repeatsUntil),
+              repeatsWhile = replaceIncludeIf(s.repeatsWhile),
+              pages = s.pages.map(page =>
+                page.copy(
+                  includeIf = replaceIncludeIf(page.includeIf),
+                  fields = replaceFields(page.fields),
+                  redirects = replaceRedirects(page.redirects),
+                  confirmation = replaceConfirmation(page.confirmation)
+                )
+              ),
+              fields = replaceFieldsNel(s.fields),
+              cyaPage = s.cyaPage.map(replaceCheckYourAnswersPage),
+              caption = newCaption
+            )
+        }
+
```

Comment:
Can we update only a caption and then call `updateClassicSection` to reduce duplication?